corehq/pillows/domain.py

```diff
@@ -17,7 +17,7 @@
 def transform_domain_for_elasticsearch(doc_dict):
     doc_ret = copy.deepcopy(doc_dict)
     sub = Subscription.objects.filter(subscriber__domain=doc_dict['name'], is_active=True)
-    doc_ret['deployment'] = doc_dict.get('deployment', None) or {}
+    doc_ret['deployment'] = doc_ret.get('deployment', None) or {}
```

Comment:
As I mentioned in the description, if I called this function with the same dict twice then you get the exception because the original dict get's modified (by the line that I changed).

I don't see how this was happening in prod so it may not fix the issue but this is still a good change.